api/src/main/java/ru/urururu/sanity/api/SourceRangeFactory.java

```diff
@@ -0,0 +1,10 @@
+package ru.urururu.sanity.api;
+
+import ru.urururu.sanity.api.cfg.SourceRange;
+
+/**
+ * @author <a href="mailto:dmitriy.g.matveev@gmail.com">Dmitry Matveev</a>
+ */
+public abstract class SourceRangeFactory<I> {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Convert the abstract class "SourceRangeFactory" into an interface. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1610)